app/controllers/api/v1/application_users_controller.rb

```diff
@@ -135,6 +135,32 @@ def index
 #end
 
   ###############################################################
+  # username
+  ###############################################################
+
+  api :GET, '/application_users/username/:username',
+            'Gets a single User with the specified username.'
+  description <<-EOS
+      All users of OpenStax have an associated User object.  This endpoint
+      allows querying additional data on a user when all that is known is the
+      User's username.
+
+      Admins (for Accounts only) are identified by the is_administrator boolean.
+      Some additional user information can be found in associations, such as
+      email addresses in ContactInfos and the password hash in Identity.
+
+      Users have the following String attributes:
+      username, first_name, last_name, full_name, title
+
+    #{json_schema(Api::V1::UserSearchRepresenter, include: :readable)}
+  EOS
+  def username
+    OSU::AccessPolicy.require_action_allowed!(:search, current_api_user, User)
+    outputs = SearchUsers.call("username:#{params[:username]}", {exact: true}).outputs
```

Comment:
I think we should be using `ApplicationUser` here instead of `User`.  An `ApplicationUser` is a wrapper around `User` dedicated for the use of one of our applications.  

You can probably delegate some of the work to the `standard_read` method in `openstax_api` (https://github.com/openstax/openstax_api/blob/master/lib/openstax/api/roar.rb#L38-L40):

```
def username
  standard_read(ApplicationUser.find_by_username(params[:username]))
end
```

You'll have to make sure the access policy for `ApplicationUser` has an appropriate check for the `:read` action.